Userland/Libraries/LibWeb/Layout/InlineFormattingContext.cpp

```diff
@@ -389,7 +389,8 @@ void InlineFormattingContext::generate_line_boxes(LayoutMode layout_mode)
                 item.margin_end,
                 item.width,
                 text_node.computed_values().line_height(),
-                move(item.glyph_run));
+                move(item.glyph_run),
+                item.direction);
```

Comment:
You should inherit direction from parent . Are you sure this is correct?

Reply:
Yeah, this logic was a bit off, we need to know both directions. The layout direction is now passed into the line builder on construction, and each glyph run contains its own direction value.